sentry-log4j2/src/main/java/io/sentry/log4j2/SentryAppender.java

```diff
@@ -0,0 +1,220 @@
+    if (throwableInformation != null) {
+      event.setThrowable(throwableInformation.getThrowable());
+    }
+
+    if (loggingEvent.getThreadName() != null) {
+      event.setExtra("thread_name", loggingEvent.getThreadName());
+    }
+
+    final Map<String, String> mdcProperties =
+        CollectionUtils.shallowCopy(loggingEvent.getContextData().toMap());
+    if (!mdcProperties.isEmpty()) {
+      event.getContexts().put("MDC", mdcProperties);
+    }
+
+    return event;
+  }
+
+  private @NotNull List<String> toParams(@Nullable Object[] arguments) {
+    if (arguments != null) {
+      return Arrays.stream(arguments)
+          .filter(Objects::nonNull)
+          .map(Object::toString)
+          .collect(Collectors.toList());
+    } else {
+      return Collections.emptyList();
+    }
+  }
+
+  /**
+   * Creates {@link Breadcrumb} from log4j2 {@link LogEvent}.
+   *
+   * @param loggingEvent the log4j2 event
+   * @return the sentry breadcrumb
+   */
+  private @NotNull Breadcrumb createBreadcrumb(final @NotNull LogEvent loggingEvent) {
+    final Breadcrumb breadcrumb = new Breadcrumb();
+    breadcrumb.setLevel(formatLevel(loggingEvent.getLevel()));
+    breadcrumb.setCategory(loggingEvent.getLoggerName());
+    breadcrumb.setMessage(loggingEvent.getMessage().getFormattedMessage());
+    return breadcrumb;
+  }
+
+  /**
+   * Transforms a {@link Level} into an {@link SentryLevel}.
+   *
+   * @param level original level as defined in log4j.
+   * @return log level used within sentry.
+   */
+  private static @NotNull SentryLevel formatLevel(@NotNull Level level) {
+    if (level.isMoreSpecificThan(Level.ERROR)) {
+      return SentryLevel.ERROR;
+    } else if (level.isMoreSpecificThan(Level.WARN)) {
+      return SentryLevel.WARNING;
+    } else if (level.isMoreSpecificThan(Level.INFO)) {
+      return SentryLevel.INFO;
+    } else {
+      return SentryLevel.DEBUG;
+    }
+  }
+
+  private @NotNull SdkVersion createSdkVersion(@NotNull SentryOptions sentryOptions) {
+    SdkVersion sdkVersion = sentryOptions.getSdkVersion();
+
+    if (sdkVersion == null) {
+      sdkVersion = new SdkVersion();
+    }
+
+    sdkVersion.setName(BuildConfig.SENTRY_LOG4J2_SDK_NAME);
+    final String version = BuildConfig.VERSION_NAME;
+    sdkVersion.setVersion(version);
+    sdkVersion.addPackage("maven:sentry-log4j2", version);
+
+    return sdkVersion;
+  }
+
+  public void setDsn(final @Nullable String dsn) {
+    this.dsn = dsn;
+  }
+
+  public void setMinimumBreadcrumbLevel(final @Nullable Level minimumBreadcrumbLevel) {
+    if (minimumBreadcrumbLevel != null) {
+      this.minimumBreadcrumbLevel = minimumBreadcrumbLevel;
+    }
+  }
+
+  public void setMinimumEventLevel(final @Nullable Level minimumEventLevel) {
+    if (minimumEventLevel != null) {
+      this.minimumEventLevel = minimumEventLevel;
+    }
+  }
+
+  @ApiStatus.Internal
+  void setTransport(@Nullable ITransport transport) {
```

Comment:
Dow e really need this or can we use the client here instead?
I assume it's for testing.
We could take a IHub in the constructor instead.
An internal overload would take it. The public overload calls into it with HubAdapter.getInstance for the prod use case

Reply:
yeah I agree with @bruno-garcia 
Sentry.init already overwrites the `NoOpTransport` to `HttpTransport` on Sdk init.
if it's for the sake of testing, I'd use the hub, I guess the other integrations also have references for transport and I'd refactor as well unless there's a need that I'm not aware of.

Reply:
The problem is that some properties are set in `MainEventProcessor` - like sdk version. This is why I moved to mocking transport instead of Hub - some things are just not possible to test when just `Hub` is mocked.

Reply:
I see, makes sense.
whats about creating a new Hub, passing options, but mocking the transport in the options, it'd be pretty much the same, but the entry point here would be the `Hub` still as every other integration.

Reply:
I have nothing against but I also don't see how adding this one level of indirection makes anything better..?

Reply:
we could discuss what would make more sense here, but that's how it is in all the others tests, we never make a reference directly to the transport, but rather the entry point, and mock things in the options if necessary, so we could assert and verify directly from them.
if we decide that this would make more sense (assert on the transport), I'd rather change everywhere, I just would like to avoid a mix of different ways of doing things.

maybe @bruno-garcia  has opinions.

Reply:
Some information like SDK version is added later, in `MainEventProcessor` - this is why i need to mock transport on lower level.

I started with mocking `Hub`, but this does not give me tools to test integration properly.